### PR TITLE
feat: [ANDROAPP-3463] Display indicators in Events from event programs

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3463)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/TrackerAnalyticsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.helpers.UidsHelper
+import org.hisp.dhis.android.core.program.ProgramIndicator
+import org.hisp.dhis.android.core.program.ProgramRuleActionType
+import org.hisp.dhis.rules.models.RuleActionDisplayKeyValuePair
+import org.hisp.dhis.rules.models.RuleActionDisplayText
+import org.hisp.dhis.rules.models.RuleEffect
+import timber.log.Timber
+
+class TrackerAnalyticsRepository(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/TrackerAnalyticsRepository.kt:28]: Do you think that we could add tests for this class?

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/IndicatorsPresenter.kt [hunk 2]
```diff
@@ -1,207 +1,25 @@
...
-                        ruleAction.location()
-                    )
-                    indicators.add(indicator)
-                }
-            }
-        }
-
-        return indicators
-    }
-
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/IndicatorsPresenter.kt:204]: Great job in this class :clap:

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventRuleEngineRepository.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.hisp.dhis.rules.RuleEngine;
+import org.hisp.dhis.rules.models.RuleDataValue;
+import org.hisp.dhis.rules.models.RuleEffect;
+import org.hisp.dhis.rules.models.RuleEvent;
+
+import java.util.List;
+
+import io.reactivex.Flowable;
+
+public final class EventRuleEngineRepository implements RuleEngineRepository {
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventRuleEngineRepository.java:22]: Do you think that we could add tests for this class?

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/EventIndicatorRepository.kt [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.helpers.UidsHelper
+import org.hisp.dhis.android.core.program.ProgramIndicator
+import org.hisp.dhis.android.core.program.ProgramRuleActionType
+import org.hisp.dhis.rules.models.RuleActionDisplayKeyValuePair
+import org.hisp.dhis.rules.models.RuleActionDisplayText
+import org.hisp.dhis.rules.models.RuleEffect
+import timber.log.Timber
+
+class EventIndicatorRepository(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/EventIndicatorRepository.kt:25]: Do you think that we could add tests for this class?

**Review by mmmateos — APPROVED** [review]

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventRuleEngineRepository.java (hunk 3 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventRuleEngineRepository.java:22]: done! :)

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/EventIndicatorRepository.kt (hunk 4 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/EventIndicatorRepository.kt:10]: done!

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/TrackerAnalyticsRepository.kt (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/TrackerAnalyticsRepository.kt:13]: done!

**Review by andresmr — APPROVED** [review]

**Pull request merged**